backend/OrigamArchitect/ArchitectWorkbench.cs

```diff
@@ -1999,7 +1999,7 @@ private T CreatePad<T>()
 		/// </summary>
 		private void InitializeConnectedServices()
 		{
-			ServiceManager.Services.AddService(new ServiceAgentFactory());
+			ServiceManager.Services.AddService(new ServiceAgentFactory(externalAgent => new ExternalAgentWrapper(externalAgent)));
```

Comment:
Why does this have to be in a constructor?

Reply:
ExternalAgentWrapper is defined in Origam.Workflow because it extends AbstractServiceAgent. The ServiceAgentFactory is defined in Origam.Workbench.Services. Instantiating the ExternalAgentWrapper  in the ServiceAgentFactory directly would require referencing Origam.Workbench.Services from ServiceAgentFactory and that would lead to a circular dependency.

Reply:
Wouldn't it be correct to do it the "old style" dependency injection, i.e. creating a "service" for it? Or should we rather do it the "constructor way" so when we redo the dependency injection sometimes later, it will be less work? @washibana

Reply:
I'd prefer to start with "constructor way" as soon as possible. It will help us to think about solutions of new problems in suitable way for transfer to DI.

Reply:
the "old style" is not dependency injection. It is Service locator "anti" pattern. Passing the factory function into the constructor is getting us closer to the dependency injection I think. And I agree we should get there sooner than later.